src/main/ruby/truffleruby/core/random.rb

```diff
@@ -66,10 +66,18 @@ def random(limit)
   end
 
   def random_range(limit)
-    min, max = limit.max.coerce(limit.min)
+    return nil if Comparable.compare_int(limit.end <=> limit.begin) < 0
+    max = -> do
+      return limit.end unless limit.exclude_end?
+
+      limit.end - 1
```

Comment:
Is that correct for floats? `rand(0.0...1.0)` returns results between 0.0 and I believe `1.0.prev_float`.

It would be good to add a spec for this case, if there isn't already.